src/executor.ts

```diff
@@ -12,12 +12,8 @@ export class Executor {
         do {
             startCandidates = Executor.getStartCandidates(jobs, stages, potentialStarters, argv.manual);
             if (startCandidates.length > 0) {
-                await PromisePool
-                    .withConcurrency(argv.concurrency ?? startCandidates.length)
-                    .for(startCandidates)
```

Comment:
I looked into the documentation of "@supercharge/promise-pool", and does seems to support throwing on promise rejections by adding:
```
.handleError(async (error) => { throw error; })
```
So, we should decide which library will be better.

Reply:
p-map is made by https://github.com/sindresorhus and has zero dependencies, it's fine 👍